### DISCUSSION TITLE
How to do paging in Darker?

### DISCUSSION THREAD

**Discussion by provanguard** [BrighterCommand/Brighter#2024] [General]
Hello,
Could you please provide some information about how to do paging using Darker?
I have been using Brighter for commands and Darker for queries for some years now, and now I would like to add paging support to my application for queries,

**Comment by DevJonny** [comment]
I meant to reply to this earlier, and then never got around to it so sorry about that @provanguard.

Paging in Darker is completely up to you, as it is not part of Darker itself as how it is done will vary depending on how you're accessing the database. One approach it to use encapsulate the paging requirements, and action in a class that you pass to your implementation of IQuery<T>.

So your paging class would look something like:

```
public class Paging
{
    public int PageSize { get; }
    public int ItemsToSkip { get; }

    public Paging (int pageSize, int itemsToSkip)
    {
        PageSize = pageSize;
        ItemsToSkip = itemsToSkip;
    }

    public IQueryable<T> Apply<T>(IQueryable<T> queryable, int additionalItems = 0)
    {
        // If you're using NHiberate or Linq based querying
       return queryable.Skip(ItemsToSkip).Take(PageSize + additionalItems);
    }
}
```

Then in your implmentation of `QueryHandler<TQuery, TResult>` you can access this, and call `Apply`.

**Reply by provanguard** [reply]
It is good to hear that it is possible to do. Knowing this, I can go in and try to implement paging. I may come back with additional questions when I get into the details. Many thanks for the reply!

**Reply by provanguard** [reply]
I am using MongoDB by the way and we have our internal libraries for DB operations on top of the Mongo C# driver.

**Reply by DevJonny** [reply]
Happy to answer any follow ups.
